scripts/coverage.sh

```
@@ -0,0 +1,61 @@
+# build as plain Python instead of Cython extensions, since coverage.py can't
+# trace into compiled extensions. Cython-only modules with no .py fallback
+# (obj_parser, numpy_parser, row_parser, ...) are not built at all in this mode
+# and are therefore not measured by this script.
+#
+# Deliberately not `set -e`: a failing test must not skip report generation
+# below, or a broken test leaves no coverage output at all to diagnose it
+# with. Each test invocation instead records its own failure into $status,
+# and the script exits with that status only after combine/report/html/xml
+# have run.
+set -uo pipefail
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)